eventstore/query_json.go

```diff
@@ -0,0 +1,232 @@
+package eventstore
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+
+	dsquery "github.com/ipfs/go-datastore/query"
+)
+
+type JSONQuery struct {
+	Ands []JSONCriterion
+	Ors  []JSONQuery
+	Sort JSONSort
```

Comment:
Sounds from the title like this PR doesn't address sorting, but since this is here I'll just leave a thought... Would be cool to specify an array of sort objects for secondary etc sorting.

Reply:
Sorry, the title is wrong. After creating the PR I added the sorting feature.

Regarding your advice, yes, I created a GH some weeks ago to add the multilevel sorting. #84

Reply:
Sweet!